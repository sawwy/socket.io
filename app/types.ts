export type UserType = {
  isOnline: boolean;
  lastSeen: Date;
  joined: Date;
  username: string;
  avatar: string;
  description: string;
};

export interface UserResponseType
  extends Omit<UserType, "lastSeen" | "joined"> {
  lastSeen: string;
  joined: string;
}

export type SelectedUserDataType = {
  user: UserType;
  rect: DOMRect;
};
