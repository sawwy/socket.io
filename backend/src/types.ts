import { MessageTypeEnum } from "./enums.ts";

export interface IUser {
  id: string;
  username: string;
  isOnline: boolean;
  lastSeen: Date;
  joined: Date;
}

export interface IUserMessage {
  messageType: MessageTypeEnum.UserMessage;
  message: string;
  user: IUser;
  timestamp: string;
}

export interface IUsers {
  [userId: string]: IUser;
}
