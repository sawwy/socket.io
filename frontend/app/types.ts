export interface IUser {
  id: string;
  isOnline: boolean;
  lastSeen: Date;
  joined: Date;
  username: string;
  avatar?: string;
  description?: string;
}

export interface UserResponseType extends Omit<IUser, "lastSeen" | "joined"> {
  lastSeen: string;
  joined: string;
}

export type SelectedUserDataType = {
  user: IUser;
  rect: DOMRect;
};

export interface IMessage {
  message: string;
  username: string;
  timestamp: string;
}
