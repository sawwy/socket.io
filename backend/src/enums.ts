export enum MessageTypeEnum {
  UserMessage = "userMessage",
  SystemMessage = "systemMessage",
}
