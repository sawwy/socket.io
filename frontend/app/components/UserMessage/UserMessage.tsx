import { IUser, IUserMessage } from "~/types";
import styles from "./styles.module.css";
import { Avatar } from "../Avatar/Avatar";
import { getMessageTime } from "~/utils/textUtils";

type UserMessagePropsType = {
  user?: IUser;
  message: IUserMessage;
};

export const UserMessage = ({ user, message }: UserMessagePropsType) => {
  const date = new Date(message.timestamp);
  console.log("date", date);

  return (
    <div className={styles.message}>
      {user && <Avatar user={user} showStatus={false} />}
      <div className={styles.messageDetails}>
        <div className={styles.messageHeader}>
          <p className={styles.username}>{message.username}</p>
          <p className={styles.messageTime}>{getMessageTime(date)}</p>
        </div>
        <p>{message.message}</p>
      </div>
    </div>
  );
};
