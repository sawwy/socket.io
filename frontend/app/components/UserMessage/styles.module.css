.message {
  display: flex;
  gap: 1rem;
  color: rgb(210, 213, 217);
}

.messageDetails {
  display: flex;
  flex-direction: column;
  gap: 0.5rem;
}

.messageHeader {
  display: flex;
  gap: 0.5rem;
  align-items: center;
}

.username {
  color: rgb(235, 236, 239);
}

.messageTime {
  font-size: 0.8em;
  color: rgb(157, 157, 162);
}
