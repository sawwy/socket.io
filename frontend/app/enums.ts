export enum UsernameErrors {
  TooShort = "Username must be longer than 3 characters",
}
