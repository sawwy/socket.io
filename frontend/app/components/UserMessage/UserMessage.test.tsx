import renderer from "react-test-renderer";
import { UserMessage } from "./UserMessage";
import { ISystemMessage, IUser, IUserMessage } from "~/types";
import { MessageTypeEnum } from "~/enums";

const testUser: IUser = {
  id: "1234-asdf",
  isOnline: true,
  lastSeen: new Date("2024-04-02T16:00:00"),
  joined: new Date("2023-03-01T08:00:00"),
  username: "qwerty",
};

const message: IUserMessage = {
  type: MessageTypeEnum.UserMessage,
  message: "Qwerty has joined the lobby",
  user: testUser,
  timestamp: new Date("2024-04-02T16:00:00").toISOString(),
};

describe("SystemMessage Snapshot", () => {
  it("Matches DOM Snapshot", () => {
    const domTree = renderer.create(<UserMessage message={message} />).toJSON();
    expect(domTree).toMatchSnapshot();
  });
});
