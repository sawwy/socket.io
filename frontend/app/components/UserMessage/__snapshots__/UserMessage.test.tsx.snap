// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SystemMessage Snapshot Matches DOM Snapshot 1`] = `
<div
  className="message"
  data-testid="chat-usermessage"
>
  <div
    className="messageDetails"
  >
    <div
      className="messageHeader"
      data-testid="chat-usermessage-header"
    >
      <p
        className="username"
      >
        qwerty
      </p>
      <p
        className="messageTime"
      >
        Today at 16:00
      </p>
    </div>
    <p>
      Qwerty has joined the lobby
    </p>
  </div>
</div>
`;
